R/zchunk_L202.Ccoef.R

```diff
@@ -1,72 +1,104 @@
              FILE = "emissions/A_PrimaryFuelCCoef",
-             FILE = "emissions/fuel_to_Ccoef",
+             FILE = "emissions/mappings/fuel_to_Ccoef",
              "L102.Ccoef_kgCGJ_F_Yh",
              "L102.Ccoef_kgCGJ_R_F_Yh"))
   } else if(command == driver.DECLARE_OUTPUTS) {
     return(c("L202.CarbonCoef"))
   } else if(command == driver.MAKE) {
 
+    # Silence package checks
+    region <- PrimaryFuelCO2Coef.name <- year <- fuel <- value <-
+      PrimaryFuelCO2Coef <- NULL
+
     all_data <- list(...)[[1]]
 
     # Load required inputs
     GCAM_region_names <- get_data(all_data, "common/GCAM_region_names")
     A_PrimaryFuelCCoef <- get_data(all_data, "emissions/A_PrimaryFuelCCoef")
-    fuel_to_Ccoef <- get_data(all_data, "emissions/fuel_to_Ccoef")
+    fuel_to_Ccoef <- get_data(all_data, "emissions/mappings/fuel_to_Ccoef")
     L102.Ccoef_kgCGJ_F_Yh <- get_data(all_data, "L102.Ccoef_kgCGJ_F_Yh")
     L102.Ccoef_kgCGJ_R_F_Yh <- get_data(all_data, "L102.Ccoef_kgCGJ_R_F_Yh")
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+    # Carbon contents of fuels in all regions
+    A_PrimaryFuelCCoef %>%
+      write_to_all_regions(c(LEVEL2_DATA_NAMES[["CarbonCoef"]]), GCAM_region_names = GCAM_region_names, has_traded = TRUE) -> L202.CarbonCoef
+
+
+    # For regions with no agricultural and land use sector (Taiwan),
+    # need to remove the passthrough supplysectors for first-gen biofuels
+    ag_en <- c("regional corn for ethanol", "regional sugar for ethanol", "regional biomassOil")
+    L202.CarbonCoef %>%
+      filter(!region %in% aglu.NO_AGLU_REGIONS | !PrimaryFuelCO2Coef.name %in% ag_en) -> L202.CarbonCoef
+
+    # if emissions.USE_GCAM3_CCOEFS = 1, using exogenous C coefs from GCAM 3.0
+    if(emissions.USE_GCAM3_CCOEFS == 1) {
+      L202.CarbonCoef <- L202.CarbonCoef
+    }
+    else if(emissions.USE_GCAM3_CCOEFS == 0) {
+      L202.CarbonCoef %>%
+        left_join(fuel_to_Ccoef, by = "PrimaryFuelCO2Coef.name") -> L202.CarbonCoef
+      if(emissions.USE_GLOBAL_CCOEFS == 1) {
+        # Using global average emissions coefficients based on CDIAC inventory and IEA energy balances
+        L102.Ccoef_kgCGJ_F_Yh %>%
+          filter(year %in% HISTORICAL_YEARS) %>%
+          filter(year == emissions.INVENTORY_MATCH_YEAR) -> L202.Ccoef_kgCGJ_F_fby
```

Comment:
Do we need both of these filters?

Reply:
Hmm. Either way, I don't think the first test is necessary; if it's not failing now, it won't fail without the HISTORICAL_YEARS line.

Reply:
@bpbond ok I'll remove that and then it looks like the rebuild worked and we're passing tests on here!